pkgs/by-name/hp/hpipm/package.nix

```diff
@@ -29,6 +29,13 @@ stdenv.mkDerivation (finalAttrs: {
     "-DBLASFEO_PATH=${blasfeo}"
   ] ++ lib.optionals (!stdenv.isx86_64) [ "-DTARGET=GENERIC" ];
 
+  postPatch = ''
+    # build shared lib
+    substituteInPlace CMakeLists.txt --replace-fail \
+      "add_library(hpipm " \
+      "add_library(hpipm SHARED "
+  '';
```

Comment:
Hmm it would be nice if upstream would have supported that naturally. Perhaps something a bit more elegant could be done with [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)?

Reply:
Hum, I packaged this looking at the CMakeLists.txt of 0.1.3, but in 0-unstable-2024-07-30 it was greatly improved. BUILD_SHARED_LIBS is already there :),